tests/template/test_com_template.py

```diff
@@ -0,0 +1,32 @@
+from libertem.web.notebook_generator.code_template import CodeTemplate
+
+com_default = '''
+com_analysis = ctx.create_com_analysis(dataset=ds, cx=125, cy=125, mask_radius=62.5)
+roi = com_analysis.get_roi()
+udf = com_analysis.get_udf()
+com_result = ctx.run_udf(ds, udf, roi, progress=True)
+com_result = com_analysis.get_udf_results(com_result, roi)
+print(com_result)
+'''
+com_plot = '''
+fig, axes = plt.subplots()
+axes.set_title("field")
+axes.imshow(com_result.field.visualized)
+fig, axes = plt.subplots()
+axes.set_title("magnitude")
+axes.imshow(com_result.magnitude.visualized)
+fig, axes = plt.subplots()
+axes.set_title("curl")
+axes.imshow(com_result.curl.visualized)
+'''
```

Comment:
> For that reason it is essential to actually run the generated notebooks and verify that they pass without errors.

Thank you, Now I have a better understanding of the testing.

>Did you have a look at https://deeplearning.lipingyang.org/2018/03/28/run-jupyter-notebook-script-from-terminal/ already?

I think `nbconvert` would be good :+1:

Reply:
> If possible, tests should also verify that the notebook generates the intended result

For implementing this, as you suggested an extra cell for saving the result would be easy. `nbconvert` doesn't offer a way to access the result from notebook. If there is no other problem I will add this in the next iteration.

Reply:
> `nbconvert` doesn't offer a way to access the result from notebook.

In the test environment you can use the fixture `tmpdir_factory` to generate temporary directories. In `conftest.py` you can see it in action, for example. One can then generate a directory from that fixture and add a cell to the notebook that explicitly saves relevant results to files in that directory, for example using [numpy.save()](https://numpy.org/doc/stable/reference/generated/numpy.save.html#numpy.save). In the test routine you can then just load the file contents from that location and compare them with the result of the analysis. That way you don't have to access variables within the notebook.

Since that is (hopefully) not too difficult and at the same time very important to verify the correctness of the generated notebooks, perhaps you can give it a try for this iteration?